Author: Luca Boccassi <luca.boccassi@gmail.com>
Description: Change NVKMS_USECS_TO_JIFFIES to pass unsigned long long to do_div
 In kernel builds from 4.6 the compiler flags are more restrictive, and now
 Werror=incompatible-pointer-types is passed, which causes the build to fail on
 32bit. do_div expects a 64 bit unsigned integer, but the NVKMS_USECS_TO_JIFFIES
 macro passes an unsigned long. Use unsigned long long instead, and cast down
 the result before returning. This macro is passed to time_mod which takes an
 unsigned long as a parameter, so casting down is fine.

--- a/nvidia-modeset/nvidia-modeset-linux.c
+++ b/nvidia-modeset/nvidia-modeset-linux.c
@@ -49,9 +49,9 @@
  */
 static inline unsigned long NVKMS_USECS_TO_JIFFIES(NvU64 usec)
 {
-    unsigned long result = usec * HZ;
+    unsigned long long result = usec * HZ;
     do_div(result, 1000000);
-    return result;
+    return (unsigned long)result;
 }
 
 
